spec/example_app/app/dashboards/log_entry_dashboard.rb

```diff
@@ -6,7 +6,12 @@ class LogEntryDashboard < Administrate::BaseDashboard
   ATTRIBUTE_TYPES = {
     id: Field::Number,
     action: Field::String,
-    logeable: Field::Polymorphic.with_options(classes: [Customer, ::Order]),
+    logeable: Field::Polymorphic.with_options(
+      classes: [
+        { klass: Customer, scope: :cn },
+        ::Order,
+      ],
+    ),
```

Comment:
I'm thinking this could be a bit more flexible if it accepted lambdas instead:

```suggestion
    logeable: Field::Polymorphic.with_options(
      classes: [
        ->{ Customer.cn },
        ::Order,
      ],
    ),
```

Then `candidate_resources_for` can check if the `klass` is a proc, and apply `.call` before `order`/`all`. What do you think?

Reply:
Good point ! Ill push a commit !

Reply:
Hi @DavidGeismarLtd, this was a while ago so I guess you might have moved on by now, but I was wondering if you made any progress with the scope. It would be nice functionality - would fix the limitation with ordering (https://github.com/thoughtbot/administrate/issues/1989) and allow filtering large record sets.

Reply:
Hey @cheeseweasel, Ill try to have a look this weekend. But if you want to finish the implementation yourself, it's fine by me  :).